File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.js
/**
 * @param {string} s
 * @return {number}
 */
var maxDepth = function(s) {
    let count = 0;
    let max = 0;
    
    for(let ch of s){
        if(ch == "("){
            count++;
            if(count > max) max = count;
        } else if(ch == ")"){
            count--;
        }   
    }
    return max;
};

File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        stack<char>st;
        int count = 0;
        int max = 0;
        
        for(int i = 0; i < s.size(); i++){
            if(s[i] != '(' && s[i] != ')'){
                continue;
            }
            
            if(s[i] == '('){
                st.push(s[i]);
                max++;
                if(max > count) count = max;
            } else if(s[i] == ')'){
                st.pop();
                max--;
            }
        }
        
        return count;
    }
};